specs/latest/2.0/index.html

```diff
@@ -1216,6 +1219,77 @@ <h2 class="no-toc">Table of contents</h2>
+      <dd>
+        Reads back data asynchronously from the bound WebGLBuffer into <code>dstBuffer</code>.
+        <br><br>
+        Let <code>buf</code> be the buffer bound to <code>target</code>.
+        If <code>length</code> is 0, let <code>copyLength</code> be
+        <code>dstBuffer.length - dstOffset</code>; otherwise, let
+        <code>copyLength</code> be <code>length</code>.
+        <br><br>
+        Copy <code>copyLength</code> typed elements (each of size <code>dstBuffer.BYTES_PER_ELEMENT</code>)
+        from <code>buf</code> into <code>dstBuffer</code>,
+        reading <code>buf</code> starting at byte index <code>srcByteOffset</code> and
+        writing into <code>dstBuffer</code> starting at element index <code>dstOffset</code>.
+        If <code>copyLength</code> is 0, no data is written to <code>dstBuffer</code>.
+        <ul>
+          <li>If no WebGLBuffer is bound to <code>target</code>,
+              an <code>INVALID_OPERATION</code> error is generated.
+          <li>If <code>target</code> is <code>TRANSFORM_FEEDBACK_BUFFER</code>,
+              and any transform feedback object is currently active,
+              an <code>INVALID_OPERATION</code> error is generated.
+          <li>If <code>dstOffset</code> is greater than <code>dstBuffer.length</code>,
+              an <code>INVALID_VALUE</code> error is generated.
+          <li>If <code>dstOffset + copyLength</code> is greater than <code>dstBuffer.length</code>,
+              an <code>INVALID_VALUE</code> error is generated.
+          <li>If <code>srcByteOffset</code> is less than zero,
+              an <code>INVALID_VALUE</code> error is generated.
+          <li>If <code>srcByteOffset + copyLength*dstBuffer.BYTES_PER_ELEMENT</code>
+              is larger than the length of <code>buf</code>,
+              an <code>INVALID_OPERATION</code> is generated.
+        </ul>
+        If any error is generated, no data is written to <code>dstBuffer</code>.
+        These errors are generated synchronously and are visible immediately after
+        <code>getBufferSubDataAsync</code> returns.
+
+        <br><br>
+        This command executes serially with respect to others in the command stream, but the data is
+        only written to <code>dstData</code> when the promise is fulfilled; it is available in
+        the Promise's success callback, but not earlier.
+
+        <br><br>
+        If a failure occurs that prevents the promise from being fulfilled, then the returned
+        Promise is rejected with an "InvalidStateError" DOMException, and no data is written to
+        <code>dstData</code>. If any of the above errors occur, the Promise is already rejected upon
+        return. If the context is lost before the Promise is fulfilled, or if <code>dstData</code>
+        is neutered at the time the Promise is fulfilled, then the Promise will be rejected later.
+        In this case, no WebGL error is generated. If the promise rejects, no data is written to
```

Comment:
@kenrussell Just want to make sure that this is the semantic we want. Right now no WebGL error will ever be generated upon resolution of the promise.